### ISSUE TITLE
[Suggestion] Warning about having "unsaved" new related entries

### ISSUE THREAD

**Issue by WoLfulus** [directus/next#844]
When creating inline items, adding a new item doesn't seem to immediately save it to the database. This made me loose some items after closing the window, since they were displayed in the interface I assumed they were registered.

Maybe adding a different display to the "unsaved" items or a warning saying it has pending modifications?

![URXhareBhH](https://user-images.githubusercontent.com/412397/97880401-e3dbc080-1cff-11eb-8842-f3285dff94aa.gif)

**Comment by rijkvanzanten** [comment]
This is the same as any other value in Directus. If you write some text in the "Nome" text-input and then close the browser, the new value for Nome isn't saved either. I think it's more confusing if some of the interfaces on the page save immediately, while others don't

**Comment by WoLfulus** [comment]
I'm not saying to immediately save the items though, just visually represent that they are not saved and/or a warning stating you must apply to create X pending items.

**Comment by benhaynes** [comment]
Similar to #843 — let's keep this consistent. I wonder if it's worth adding a `onbeforeunload` event to check and warn the user if there are unsaved changes when they try to hard-exit a page... this would mirror the notification you get when you try to soft-leave the page. I usually think those events are invasive, but they make sense here. Thoughts @rijkvanzanten @WoLfulus ?

**Comment by rijkvanzanten** [comment]
From what I can tell, @WoLfulus is asking to add an icon/ note to the m2m/o2m interface itself indicating there are unsaved changes, not a warning when exiting the page (that already exists)

**Comment by benhaynes** [comment]
Hmm, but the root issue here is:

> "This made me loose some items after closing the window"

If we have the catch when closing or navigating away from Directus, then this is solved, no? I'm not completely against an indicator, but that needs to be added in all interfaces (and we can't control custom ones), so this is a more global and "fool proof" solution.

**Comment by rijkvanzanten** [comment]
> "This made me loose some items after closing the window"

Closing the whole browser tab, yeah, that's not something we can really prevent in our own styled dialog situation.

**Comment by benhaynes** [comment]
Yeah, that's why I was saying we could "mirror" what we do... this would have to be a system/browser dialog that warns about unsaved changes so the user has a chance to cancel if it was an accident.
